crates/matrix-sdk-common/src/linked_chunk/as_vector.rs

```diff
@@ -407,6 +407,14 @@ impl UpdateToVectorDiff {
                 }
 
                 Update::Clear => {
+                    // Clean `self.chunks`.
+                    self.chunks.clear();
+
+                    // An `Update::Clear` happens when `LinkedChunk::clear` is called, which removes
+                    // all chunks, and re-adds one `Chunk` with `FIRST_IDENTIFIER` as its chunk
+                    // identifier.
+                    self.chunks.push_back((ChunkIdentifierGenerator::FIRST_IDENTIFIER, 0));
```

Comment:
I think it's the wrong layer to do that, the linked chunk should save that update and propagate it instead; see also https://github.com/matrix-org/matrix-rust-sdk/issues/4314 which describes the same issue at construction.

Reply:
Oh, sure! Good idea. I'll update this PR to address that.

Reply:
Okay, this PR has the correct approach now, but another PR is necessary to fully fix https://github.com/matrix-org/matrix-rust-sdk/issues/4314. I'm already on it, but I think it's nice to keep them separate, otherwise the diff will be a bit confusing.